split-webapp/split/package.json

```diff
@@ -6,19 +6,22 @@
   "dependencies": {
     "@material-ui/core": "^4.9.5",
     "@material-ui/icons": "^4.9.1",
+    "@material-ui/lab": "^4.0.0-alpha.48",
     "@material-ui/styles": "^4.9.0",
     "@reduxjs/toolkit": "^1.2.5",
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "@unicef/material-ui-currency-textfield": "^0.8.6",
```

Comment:
Do we really need a dependency for this? It's only used once and seems to be solely for styling purposes, which we can implement pretty easily.

Reply:
I've been using `CurrencyTextField` not really for it's styling, but for its as-you-type input formatting so the user can only enter in a valid currency value:

- No letters
- At most a single decimal point
- At most 2 digits after decimal point
- No leading zeros
- Automatic comma insertion
    - Backspace key should skip over the comma and delete the digit prior to it
    - Moving the caret with arrow keys should skip over commas
- Automatic rounding when inserting the decimal point somewhere in the middle of the number
- Parsing and fixing values after copy and paste
- Inserting two decimal places on blur

I've tried implementing something similar myself last year, and I found it quite hard to get everything right. So, for this project, I wanted to rely on a third party package to do this.

Do you reckon it's worth the dependency in this case?

Reply:
I think that's reasonable, thanks for clarifying.